File: sample-chat-obj-c/sample-chat/Components/ChatScreen/Views/ChatCollectionView/ChatNotificationCell/ChatNotificationCell.m
//
//  ChatNotificationCell.m
//  samplechat
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "ChatNotificationCell.h"

@implementation ChatNotificationCell

+ (ChatCellLayoutModel)layoutModel {
    
    ChatCellLayoutModel defaultLayoutModel = [super layoutModel];
    
    defaultLayoutModel.containerInsets = UIEdgeInsetsMake(4, 10, 4, 10);
    defaultLayoutModel.avatarSize = CGSizeMake(0, 0);
    defaultLayoutModel.topLabelHeight = 0;
    defaultLayoutModel.bottomLabelHeight = 0;
    
    return defaultLayoutModel;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.notificationLabel.backgroundColor = [UIColor clearColor];
}

@end

File: sample-chat-swift/sample-chat-swift/Components/ChatScreen/Chat/CollectionView/CollectionCells/ChatNotificationCell/ChatNotificationCell.swift
//
//  ChatNotificationCell.swift
//  Swift-ChatViewController
//
//  Created by Injoit on 1/28/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

class ChatNotificationCell: ChatCell {
    @IBOutlet weak var notificationLabel: UILabel!
    
    override class func layoutModel() -> ChatCellLayoutModel {
        let containerInsets = UIEdgeInsets(top: 4.0, left: 10.0, bottom: 4.0, right: 10.0)
        var defaultLayoutModel = super.layoutModel()
        defaultLayoutModel.containerInsets = containerInsets
        defaultLayoutModel.avatarSize = .zero
        defaultLayoutModel.timeLabelHeight = 0.0
        
        return defaultLayoutModel
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        notificationLabel.backgroundColor = .clear
    }
}
